conn.go

```diff
@@ -423,3 +430,104 @@ func (cc *Conn) allocateTransactionID() uint32 {
 	}
 	return cc.lastID
 }
+
+// getMessageSize returns the total size of all messages in the buffer.
+func (cc *Conn) getMessageSize() int {
+	var total int
+	for _, msg := range cc.messages {
+		total += len(msg.Data) + unix.NLMSG_HDRLEN
+	}
+	return total
+}
+
+// canEnlargeBuffers returns true if the connection can automatically enlarge
+// the write and read buffers of the netlink connection.
+func (cc *Conn) canEnlargeBuffers() bool {
+	// If there are sock options, we assume that the user has already set the
+	// buffers to a fixed size.
+	if len(cc.sockOptions) > 0 {
```

Comment:
This is the approach that made most sense to me because it keeps the code backwards compatible. Users will not suddenly get confused when their [explicitly defined buffers](https://github.com/google/nftables/blob/main/nftables_test.go#L7274) change in size.

The only downside I can think of is that if anyone uses `WithSockOptions` for anything other than setting the buffer sizes then they won't benefit from this feature. However, I am not sure it's often or ever the case ?

Anyways, let me know if you have any thoughts on that or an idea for an alternative approach.

Reply:
GitHub code search shows 0 usages of nftables.WithSockOptions. 

Can you add a comment to WithSockOptions that explicitly makes users aware of this? Maybe something like:

Note that when using WithSockOptions, you are responsible for providing a large-enough read and write buffer, whereas normally, the nftables package automatically enlarges the buffers as needed.